tutorials/grails3-send-push-notification-using-firebase-cloud-messaging/index.md

```diff
@@ -0,0 +1,206 @@
+author: didinj
+tags: Firebase, Firebase Cloud Messaging, FCM, Grails, Grails 3, Push Notification
+date_published: 2017-08-23
+---
+
+This tutorial shows how to send push notification using [Firebase Cloud Messaging (FCM)](https://firebase.google.com/docs/cloud-messaging) and [Grails 3 Framework](https://grails.org/) Web Application.
+
+## Objectives
+
+- Configure Firebase to send push notification via Web Application
+- Create a new Grails 3.3.0 Web Application
+- Create function for sending push notification on Grails 3.3.0
+
+## Before You Begin
+
+The following is required for this tutorial:
+
+- Terminal or Command Line
+- Text Editor or IDE ([Atom](https://atom.io/), [Netbeans](https://netbeans.org/))
+- [Java Development Kit 8](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
+- Grails 3.3.0 SDK (https://grails.org/download.html)
+- Google or Gmail account for accessing Firebase dashboard
+
+## Costs
+
+This tutorial uses Firebase Cloud Messaging service which is free for limited features and capacities.
+
+## Configure Firebase Cloud Messaging
+
+1. Open your browser then go to [Firebase](https://firebase.google.com/).
+2. Login using your Gmail account then click Go to Console button.
+3. Click add project button then give it a name. For this tutorial name it "GrailsFCM" then click Create Project button.
+4. It will redirect to the Project dashboard. Click the gear icon in the left menu and select Project Settings.
+5. Messaging tab and copy the Legacy Server Key to your notepad. You will use it later in the Grails web application.
+6. Next, configuration your app for the [Android](https://firebase.google.com/docs/cloud-messaging/android/client) or [iOS](https://firebase.google.com/docs/cloud-messaging/ios/client).
+
+## Create New Grails 3.3.0 Web Application
+
+After installing JDK 8 and Grails 3.3.0 and updating your environment path, you can create the new Grails 3.3.0 application.
+
+1. Open terminal or command line then go to your projects folder.
+2. Type this command.
+     
+   grails create-app grails-fcm
+        
+3. Go to the newly created Grails 3 application folder.
+    
+   cd grails-fcm
+    
+4. Enter Grails 3 interactive console by type this command.
+    
+   grails
+    
+5. In the Grails interactive console, you will see the following:
+    
+   | Resolving Dependencies. Please wait...
+   
+   CONFIGURE SUCCESSFUL
+
+   | Enter a command name to run. Use TAB for completion:
+   grails>
+    
+6. Test your Grails web application by typing this command in the Grails interactive console.
+    
+   run-app
+    
+## Add Grails Datastore Rest Client Dependency
+
+To access Firebase using REST API, add the Grails Datastore Client Rest dependency to the Grails 3 application.
+
+1. Open and edit build.gradle.
+
+2. Add the Grails Datastore Client Rest dependency inside the dependencies section. There are two dependencies, one inside buildscript and the other outside buildscripts.
```

Comment:
question: one inside buildscript and the other outside buildscripts.
Do you need to add "s" to the first buildscript or drop the "s" from buildscripts. Or, are we talking about 2 different scripts?